Disable building XPC services when launching WebKit processes as extensions

pvollan opened the pull request:
#### a945ee0ce3091b1884cf601b894ebc9a8d776320
<pre>
Disable building XPC services when launching WebKit processes as extensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=267248">https://bugs.webkit.org/show_bug.cgi?id=267248</a>
<a href="https://rdar.apple.com/120677588">rdar://120677588</a>

Reviewed by Elliot Williams, Brent Fulgham and Timothy Hatcher.

Disable building XPC services when launching WebKit processes as extensions. Additonally, this patch
is changing the bundle ID of the extensions to match the XPC service bundle IDs. This is to ensure
that existing bundle ID checks will continue to work. Also disable fallback to XPC service launch,
since these are not being built anymore. Instead, add a temporary fallback using the old extension
bundle IDs, since they may be cached on the system.

This patch is relanding <a href="https://commits.webkit.org/272887@main">https://commits.webkit.org/272887@main</a> and
<a href="https://commits.webkit.org/272868@main.">https://commits.webkit.org/272868@main.</a>

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUExtension.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/NetworkingExtension.xcconfig:
* Source/WebKit/Configurations/WebContentCaptivePortalExtension.xcconfig:
* Source/WebKit/Configurations/WebContentExtension.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
* Source/WebKit/UIProcess/Launcher/ProcessLauncher.h:
(WebKit::ProcessLauncher::setIsRetryingLaunch):
(WebKit::ProcessLauncher::isRetryingLaunch const):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::serviceNameAndIdentifier):
(WebKit::launchWithExtensionKit):
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/273113@main">https://commits.webkit.org/273113@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9a3d7f9f70dca06b093c9854e51b43bebe00a08b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34072 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12870 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36047 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36712 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30878 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15254 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10008 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29926 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34575 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10887 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30403 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9500 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9607 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38018 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30954 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30748 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35709 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9734 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7649 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33612 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11520 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10299 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4405 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10546 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/Scripts/copy-webcontent-resources-to-private-headers.sh:
```diff
@@ -39,7 +39,9 @@ PROCESSED_INFOPLIST="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
 UNPROCESSED_INFOPLIST="${INFOPLIST_FILE}"
 COPIED_INFOPLIST="${WEB_CONTENT_RESOURCES_PATH}/Info-WebContent.plist"
 echo "Copying Info.plist from ${UNPROCESSED_INFOPLIST} to ${COPIED_INFOPLIST}"
```
You should move this line inside the if block too.

pvollan replied:
Ah, good point, will fix.

Thanks for reviewing!

xeenon commented on Source/WebKit/Shared/AuxiliaryProcessExtensions/GPUExtension-Info.plist:
```diff
@@ -5,7 +5,7 @@
 	<dict>
 		<key>EXExtensionPointIdentifier</key>
-		<string>com.apple.webkit.gpu.extension</string>
+		<string>com.apple.web-browser-engine.gpu</string>
```
Why are these strings changing?

emw-apple replied:
I think they were always wrong, and we used to have a post-build script that would fix them up.

xeenon commented on Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
```diff
@@ -159,6 +159,19 @@ static void launchWithExtensionKit(ProcessLauncher& processLauncher, ProcessLaun
+            RELEASE_LOG_FAULT(Process, "Error launching process, description '%s', reason '%s'", String([error localizedDescription]).utf8().data(), String([error localizedFailureReason]).utf8().data());
+#if PLATFORM(IOS)
+            // Fallback to legacy extension identifiers
+            // This fallback is temporary and should be removed when possible. See rdar://120793705.
```
Make this a FIXME to remove it.

emw-apple commented on Source/WebKit/Configurations/GPUService.xcconfig:
```diff
@@ -25,6 +25,7 @@
 
 PRODUCT_NAME = com.apple.WebKit.GPU;
 PRODUCT_BUNDLE_IDENTIFIER = $(PRODUCT_NAME);
+
```
nit: delete these blank lines when you have a chance

pvollan replied:
Done!

emw-apple reviewed (changes requested):
Marking r- because we can't land this until we make internal changes to delete existing XPC services when installing roots. See https://github.com/WebKit/WebKit/pull/22643#pullrequestreview-1818767775 for my reasoning.

pvollan committed:
Disable building XPC services when launching WebKit processes as extensions
https://bugs.webkit.org/show_bug.cgi?id=267248
rdar://120677588

Reviewed by Elliot Williams, Brent Fulgham and Timothy Hatcher.

Disable building XPC services when launching WebKit processes as extensions. Additonally, this patch
is changing the bundle ID of the extensions to match the XPC service bundle IDs. This is to ensure
that existing bundle ID checks will continue to work. Also disable fallback to XPC service launch,
since these are not being built anymore. Instead, add a temporary fallback using the old extension
bundle IDs, since they may be cached on the system.

This patch is relanding https://commits.webkit.org/272887@main and
https://commits.webkit.org/272868@main.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUExtension.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/NetworkingExtension.xcconfig:
* Source/WebKit/Configurations/WebContentCaptivePortalExtension.xcconfig:
* Source/WebKit/Configurations/WebContentExtension.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
* Source/WebKit/UIProcess/Launcher/ProcessLauncher.h:
(WebKit::ProcessLauncher::setIsRetryingLaunch):
(WebKit::ProcessLauncher::isRetryingLaunch const):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::serviceNameAndIdentifier):
(WebKit::launchWithExtensionKit):
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/273113@main
